index.ts

```diff
@@ -1,35 +1,52 @@
 export class OatyArray {
-  private _original: object[]
-  private _keys: string[]
-  private _transposed: object = {}
+  private _keys: string[] | undefined
+  private _original: object[] = []
+  private _transposed: { [key: string]: any } = {}
+
+  constructor(options?: { data?: object[], keys?: string[] }) {

```

Comment:
`data` is optional again

Reply:
can you move `data` back outside of `options`?

Reply:
Not really, if I wanted to set just the `keys` I would rather call `new OatyArray({keys: ["a"]})` than `new OatyArray([], {keys: ["a"]})`.

Reply:
In most cases an `OatyArray` would be initialised with data - I prefer `new OatyArray(data, {})`

Reply:
We should discourage initialising with data, though. 
In my case, I am generating the data. It is much faster to add data when it's created than to add it all at the end. You should only pass in the data if comes from a third party source.

Reply:
> We should discourage initialising with data, though.

Disagree with this - your case isn't everyone's case

Reply:
It's not about that. 
`data` cannot be optional if it is outside of `options`.
Passing `null`, `void`, `[]`, `undefined` etc for a mandatory field is bad design; this is the exact purpose of overloads and default/optional arguments.

Reply:
I don't think it's bad design. You initialise an empty array by doing `let arr = []` and I think it makes sense for `OatyArray` to take `[]` as data for empty initialisation.

Regardless, `data` is not an 'option'; the `options` param should be reserved for… options

Reply:
I don't agree but I will revert and we can discuss this another time